### ISSUE TITLE
using static libs breaks linking

### ISSUE THREAD

**Issue by BlackLotus** [Isotr0py/pillow-jpegxl-plugin#8]
At least for me building against statically linked libs results in linking problems like so:
```
  = note: /usr/bin/ld: cannot find -ljxl: No such file or directory
          /usr/bin/ld: cannot find -ljxl_threads: No such file or directory
          /usr/bin/ld: cannot find -lhwy: No such file or directory
          /usr/bin/ld: cannot find -lbrotlidec-static: No such file or directory
          /usr/bin/ld: cannot find -lbrotlienc-static: No such file or directory
          /usr/bin/ld: cannot find -lbrotlicommon-static: No such file or directory
          collect2: error: ld returned 1 exit status
```
When changing build.rs to look more like this
```
use std::env;

fn main() {
    // Static link libjxl
    #[cfg(not(feature = "vendored"))]
    {
        println!("cargo:rustc-link-lib=jxl");
        println!("cargo:rustc-link-lib=jxl_threads");

        println!("cargo:rustc-link-lib=hwy");
        if let Ok(path) = env::var("DEP_HWY_LIB") {
            println!("cargo:rustc-link-search=native={}", path);
        }

        println!("cargo:rustc-link-lib:+whole-archive=brotlidec");
        println!("cargo:rustc-link-lib=brotlienc");
        println!("cargo:rustc-link-lib=brotlicommon");
        if let Ok(path) = env::var("DEP_BROTLI_LIB") {
            println!("cargo:rustc-link-search=native={}", path);
        }
    }

    // Dynamic link c++
    // TODO: Support MSVC and use Cargo
    let platform = env::consts::OS;
    match platform {
        // Since MSVC will stuck on building libjxl
        // Linux and Windows should all use GNU toolchain
        "linux" | "windows" => println!("cargo:rustc-link-lib=stdc++"),
        "macos" => println!("cargo:rustc-link-lib=c++"),
        _ => panic!("Not implemented c++ link on {}", platform)
    }
}
```
it builds.

I'm using ArchLinux, installed brotli and libjxl rust 1.74

**Comment by Isotr0py** [comment]
Ooops, I forgot to add `Build from source` section in readme. 

Since we use static link in `build.rs`, we need to pass `DEP_BROTLI_LIB` and `DEP_HWY_LIB` as environment variables with paths of `brotli` and `highway`.

I used dynamic link to build the `.pyd` library before, but the built `.pyd` failed to import in python. As a result, I changed it to static in `build.rs`.
 
Did the built library work on python? If it works, I will re-add it to `build.rs`.
